src/common/transformations/tests/op_conversions/einsum_decomposition_test.cpp

```diff
@@ -0,0 +1,663 @@
+        auto data_2 = std::make_shared<ov::op::v0::Parameter>(element::f32, data_shape_2);
+
+        // Transpose data_2 so that common labels, separated and reduced labels are grouped for both operands.
+        auto order_2 = ov::op::v0::Constant::create(element::i64, {3}, {0, 2, 1});
+        auto transpose_2 = std::make_shared<ov::op::v1::Transpose>(data_2, order_2);
+
+        // Broadcast data_1 and data_2 to common broadcasted shapes for common and reduced subshapes.
+        // Subgraphes are constant-folded, target subshapes are calculated broadcast_merge_shapes function.
+        auto broadcast_shape_constant_1 =
+            ov::op::v0::Constant::create(element::i64, Shape{data_shape_1.size()}, {5, 2});
+        auto broadcast_shape_constant_2 =
+            ov::op::v0::Constant::create(element::i64, Shape{data_shape_2.size()}, {10, 25, 2});
+        auto broadcast_1 = std::make_shared<ov::op::v3::Broadcast>(data_1,
+                                                                   broadcast_shape_constant_1,
+                                                                   ov::op::BroadcastType::BIDIRECTIONAL);
+        auto broadcast_2 = std::make_shared<ov::op::v3::Broadcast>(transpose_2,
+                                                                   broadcast_shape_constant_2,
+                                                                   ov::op::BroadcastType::BIDIRECTIONAL);
+        // Optionally reshape broadcasted data_1 and data_2 so separate and reduced labels are represented by one
+        // dimension. Subgraphes are constant-folded, target subshapes are calculated broadcast_merge_shapes function.
+        auto shape_constant_1 = ov::op::v0::Constant::create(element::i64, Shape{2}, {5, 2});
+        auto shape_constant_2 = ov::op::v0::Constant::create(element::i64, Shape{2}, {250, 2});
+        auto reshape_1 = std::make_shared<ov::op::v1::Reshape>(broadcast_1, shape_constant_1, false);
+        auto reshape_2 = std::make_shared<ov::op::v1::Reshape>(broadcast_2, shape_constant_2, false);
+        // Apply MatMul operation for formatted inputs.
+        auto matmul = std::make_shared<ov::op::v0::MatMul>(reshape_1, reshape_2, false, true);
+        // Optionally reshape back by unrolling dimensions corresponding to separate labels if needed.
+        // Subgraphes are constant-folded, target subshapes are calculated broadcast_merge_shapes function.
+        auto shape_out = ov::op::v0::Constant::create(element::i64, {3}, {5, 10, 25});
+        auto reshape_out = std::make_shared<ov::op::v1::Reshape>(matmul, shape_out, false);
+        // Transpose to the original order of output labels.
+        auto order_out = ov::op::v0::Constant::create(element::i64, {3}, {1, 0, 2});
+        auto transpose_out = std::make_shared<ov::op::v1::Transpose>(reshape_out, order_out);
+
+        model_ref = std::make_shared<Model>(NodeVector{transpose_out}, ParameterVector{data_1, data_2});
+    }
+}
```

Comment:
You can consider enablement of accuracy comparison to verify that the decomposed sub-graph gives the same results as the original `comparator.enable(FunctionsComparator::CmpValues::ACCURACY);`